synfig-studio/src/gui/workarea.cpp

```diff
@@ -1536,11 +1555,11 @@ WorkArea::on_drawing_area_event(GdkEvent *event)
 			double y(event->button.y), x(event->button.x);
 
 			// Erase the guides if dragged into the rulers
-			if(curr_guide_is_x && !std::isnan(x) && x<0.0 )
-				get_guide_list_x().erase(curr_guide);
+			if(curr_guide->isVertical && !std::isnan(x) && x<0.0 )
+				get_guide_list().erase(curr_guide);
 			else
-			if(!curr_guide_is_x && !std::isnan(y) && y<0.0 )
-				get_guide_list_y().erase(curr_guide);
+			if(!curr_guide->isVertical && !std::isnan(y) && y<0.0 )
+				get_guide_list().erase(curr_guide);
```

Comment:
Can't rotated guidelines be erased by dragging them back to rulers?

Reply:
Yes of course, but vertical guides (guides made from the vertical ruler), are only erased when dragged back to the vertical rulers. Same for horizontal.

Reply:
Are you saying that if:
- User created a guideline by dragging it from the vertical ruler
- Later it rotates that vertical guideline and make it horizontal
- Drag back to the horizontal ruler it won't be erased?

Reply:
done